src/formatters/base.ts

```diff
@@ -0,0 +1,322 @@
+          ? key
+          : parseInt(trimUnderscore(key), 10)
+        : key;
+      const isLast = index === length - 1;
+      fn(key, leftKey, moveDestinations[leftKey], isLast);
+    }
+  }
+
+  getDeltaType(delta, movedFrom) {
+    if (typeof delta === "undefined") {
+      if (typeof movedFrom !== "undefined") {
+        return "movedestination";
+      }
+      return "unchanged";
+    }
+    if (isArray(delta)) {
+      if (delta.length === 1) {
+        return "added";
+      }
+      if (delta.length === 2) {
+        return "modified";
+      }
+      if (delta.length === 3 && delta[2] === 0) {
+        return "deleted";
+      }
+      if (delta.length === 3 && delta[2] === 2) {
+        return "textdiff";
+      }
+      if (delta.length === 3 && delta[2] === 3) {
+        return "moved";
+      }
+    } else if (typeof delta === "object") {
+      return "node";
+    }
+    return "unknown";
+  }
+
+  parseTextDiff(value) {
+    const output = [];
+    const lines = value.split("\n@@ ");
+    for (let i = 0, l = lines.length; i < l; i++) {
+      const line = lines[i];
+      const lineOutput = {
+        pieces: [],
+      };
+      const location = /^(?:@@ )?[-+]?(\d+),(\d+)/.exec(line).slice(1);
+      // @ts-ignore
+      lineOutput.location = {
+        line: location[0],
+        chr: location[1],
+      };
+      const pieces = line.split("\n").slice(1);
+      for (
+        let pieceIndex = 0, piecesLength = pieces.length;
+        pieceIndex < piecesLength;
+        pieceIndex++
+      ) {
+        const piece = pieces[pieceIndex];
+        if (!piece.length) {
+          continue;
+        }
+        const pieceOutput = {
+          type: "context",
+        };
+        if (piece.substr(0, 1) === "+") {
+          pieceOutput.type = "added";
+        } else if (piece.substr(0, 1) === "-") {
+          pieceOutput.type = "deleted";
+        }
+        // @ts-ignore
+        pieceOutput.text = piece.slice(1);
+        lineOutput.pieces.push(pieceOutput);
+      }
+      output.push(lineOutput);
+    }
+    return output;
+  }
+
+  protected nodeBegin(
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    context: any,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    key: any,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    leftKey: any,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    type: string,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    nodeType: string,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    isLast: boolean
+  ) {
+    throw new Error("::nodeBegin Not implemented!");
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  protected rootBegin(context: any, type: string, nodeType: string) {
+    throw new Error("::rootBegin Not implemented!");
+  }
+
+  protected nodeEnd(
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
If you want, you could use `// eslint-disable @typescript-eslint/no-unused-vars` and `// eslint-enable @typescript-eslint/no-unused-vars` so that you only have two comments instead six.